src/views/collection-panel/collection-panel.tsx

```diff
@@ -0,0 +1,187 @@
+import { CollectionResource } from '../../models/collection';
+
+type CssRules = 'card' | 'iconHeader';
+
+const styles: StyleRulesCallback<CssRules> = (theme: ArvadosTheme) => ({
+    card: {
+        marginBottom: '20px'
+    },
+    iconHeader: {
+        fontSize: '1.875rem',
+        color: theme.customs.colors.yellow700
+    }
+});
+
+const MENU_OPTIONS = [
+    {
+        title: 'Edit collection',
+        icon: RenameIcon
+    },
+    {
+        title: 'Share',
+        icon: ShareIcon
+    },
+    {
+        title: 'Move to',
+        icon: MoveToIcon
+    },
+    {
+        title: 'Add to favorites',
+        icon: AddFavoriteIcon
+    },
+    {
+        title: 'Copy to project',
+        icon: CopyIcon
+    },
+    {
+        title: 'View details',
+        icon: DetailsIcon
+    },
+    {
+        title: 'Provenance graph',
+        icon: ProvenanceGraphIcon
+    },
+    {
+        title: 'Advanced',
+        icon: AdvancedIcon
+    },
+    {
+        title: 'Remove',
+        icon: RemoveIcon
+    }
+];
+
+interface CollectionPanelDataProps {
+    item: CollectionResource;
+}
+
+interface CollectionPanelActionProps {
+    onItemRouteChange: (collectionId: string) => void;
+}
+
+type CollectionPanelProps = CollectionPanelDataProps & CollectionPanelActionProps 
+                            & WithStyles<CssRules> & RouteComponentProps<{ id: string }>;
+
+export const CollectionPanel = withStyles(styles)(
+    connect((state: RootState) => ({ item: state.collectionPanel.item }))(
+        class extends React.Component<CollectionPanelProps> { 
+
+            state = {
+                anchorEl: undefined
+            };
+
+            showMenu = (event: any) => {
+                this.setState({ anchorEl: event.currentTarget });
+            }
+
+            closeMenu = () => {
+                this.setState({ anchorEl: undefined });
+            }
+
+            displayMenuAction = () => {
+                return <IconButton
+                    aria-label="More options"
+                    aria-owns={this.state.anchorEl ? 'submenu' : undefined}
+                    aria-haspopup="true"
+                    onClick={this.showMenu}>
+                    <MoreOptionsIcon />
+                </IconButton>;
+            }
+
+            render() {
+                const { anchorEl } = this.state;
+                const { classes, item } = this.props;
+                return <div>
+                        <Card className={classes.card}>
+                            <CardHeader 
+                                avatar={ <CollectionIcon className={classes.iconHeader} /> }
+                                action={ 
+                                    <IconButton
+                                        aria-label="More options"
+                                        aria-owns={anchorEl ? 'submenu' : undefined}
+                                        aria-haspopup="true"
+                                        onClick={this.showMenu}>
```

Comment:
There's a global context menu module. I think this component should have `onContextMenu` prop so it could be connected to the global context menu.

Reply:
@burzomir I will check